dask_cloudprovider/azure/utils.py

```diff
@@ -0,0 +1,178 @@
+import asyncio
+import datetime
+import logging
+
+import aiohttp
+from distributed.diagnostics.plugin import WorkerPlugin
+from tornado.ioloop import IOLoop, PeriodicCallback
+
+
+logger = logging.getLogger(__name__)
+
+
+class AzurePreemptibleWorkerPlugin(WorkerPlugin):
+    """A worker plugin for azure spot instances
+
+    This worker plugin will poll azure's metadata service for preemption notifications.
+    When a node is preempted, the plugin will attempt to shutdown gracefully all workers
+    on the node.
+
```

Comment:
It would be good to note here that this plugin can be used on any worker running on Azure Preemtible VMs, not just ones created by `dask-cloudprovider`.